test/test-prepare-check.c

```diff
@@ -0,0 +1,153 @@
+	assert(prepare_callback_1_count > check_callback_1_count);
+	assert(prepare_callback_1_count > check_callback_2_count);
+
+	/* if we've just scheduled the timeout event at the beginning of the iteration, save the current time and assert that the timeout is what we set */
+	if (start_time.tv_sec == 0) {
+		start_time = info->now;
+		assert(info->timeout->tv_sec == 1);
+		assert(info->timeout->tv_usec == 0);
+	}
+}
+
+void
+prepare_callback_2(struct event_watcher *watcher, const struct prepare_watcher_cb_info *info)
+{
+	log_prepare_callback;
+	++prepare_callback_2_count;
+
+	/* prepare_callback_2 should only fire on the first iteration, and should fire before both check callbacks */
+	assert(iteration == 0);
+	assert(prepare_callback_2_count > check_callback_1_count);
+	assert(prepare_callback_2_count > check_callback_2_count);
+}
+
+void
+check_callback_1(struct event_watcher *watcher, const struct check_watcher_cb_info *info)
+{
+	log_check_callback;
+	++check_callback_1_count;
+
+	/* check_callback_1 should always fire before check_callback_2 */
+	assert(check_callback_1_count > check_callback_2_count);
+
+	/* save the end time, in case the timeout fires this time through the event loop */
+	end_time = info->now;
+}
+
+void
+check_callback_2(struct event_watcher *watcher, const struct check_watcher_cb_info *info)
+{
+	log_check_callback;
+	++check_callback_2_count;
+
+	/* check_callback_2 should only fire on the first iteration */
+	assert(iteration == 0);
+}
+
+void
+timeout_callback(evutil_socket_t fd, short events, void *arg)
+{
+	printf("timeout_callback\n");
+
+	/* the duration between the start and end times should be at least 1 second */
+	assert(end_time.tv_sec >= start_time.tv_sec + 1);
+}
+
+int
+main(int argc, char **argv)
+{
+	struct event_base *base = event_base_new();
+
+	/* install prepare and check watchers */
+	event_watcher_prepare_new(base, &prepare_callback_1);
+	event_watcher_check_new(base, &check_callback_1);
+	struct event_watcher *prepare_callback_2_watcher = event_watcher_prepare_new(base, &prepare_callback_2);
+	struct event_watcher *check_callback_2_watcher = event_watcher_check_new(base, &check_callback_2);
+
+	/* schedule an 1 second timeout event, and run the event loop until the timeout fires */
+	printf("iteration 1\n-----------\n");
+	struct timeval tv;
+	tv.tv_sec = 1;
+	tv.tv_usec = 0;
+	event_base_once(base, -1, EV_TIMEOUT, &timeout_callback, 0, &tv);
+	event_base_dispatch(base);
+
+	/* second iteration: free two of the watchers, schedule a timeout and run the event loop again */
+	printf("iteration 2\n-----------\n");
+	iteration = 1;
+	start_time.tv_sec = 0;
+	event_watcher_free(prepare_callback_2_watcher);
+	event_watcher_free(check_callback_2_watcher);
+	event_base_once(base, -1, EV_TIMEOUT, &timeout_callback, 0, &tv);
+	event_base_dispatch(base);
+
+	/* this will free the watchers that weren't already freed in timeout_callback */
+	event_base_free(base);
+	return (0);
```

Comment:
please do not wrap the value (personally I think that this style came from some parts that had been migrated from other places, and that was a long time ago)

Reply:
so just `return 0;`